tests/valgrind-check/valgrind.sh

```diff
@@ -102,15 +110,25 @@ print_ps
 echo "Starting cf-serverd with valgrind in background:"
 valgrind $VG_OPTS --log-file=serverd.txt /var/cfengine/bin/cf-serverd --no-fork 2>&1 > serverd_output.txt &
 server_pid="$!"
-sleep 20
 
 echo "Starting cf-execd with valgrind in background:"
 valgrind $VG_OPTS --log-file=execd.txt /var/cfengine/bin/cf-execd --no-fork 2>&1 > execd_output.txt &
 exec_pid="$!"
-sleep 10
 
 print_ps
 
+# cf-serverd running under valgrind can be really slow to start, let's give it
+# some time before we move on and potentially hit the wall if it's actually
+# malfunctioning
+tries=12 # max 2 minutes
+while /var/cfengine/bin/cf-net -H $BOOTSTRAP_IP connect | grep Failed; do
+  tries=$((tries - 1))
+  if [ $tries -le 0 ]; then
+    break;
+  fi
+  sleep 10
+done
```

Comment:
Any reason for doing this after starting cf-execd? I guess the reasoning for how it was before (cf-serverd, sleep, cf-execd) was to let cf-serverd start properly before cf-execd has a chance to mess with it.

Reply:
Why would cf-execd mess with cf-serverd?

Reply:
The logic is to do everything else we can before we potentially `sleep`.

Reply:
@vpodzime `cf-execd` starts all the CFEngine services, right? (indirectly via cf-agent, systemd, or other ways). At least intuitively, it feels like `valgrind + cf-serverd` starting slowly could cause different situations to occur (race conditions) when `valgrind + cf-execd` is started immediately after it. I don't know how realistic / probable it is, though.

Reply:
> `cf-execd` starts all the CFEngine services, right?

Not in case of systemd, we use a path condition on `policy_server.dat` IIRC. But on non-systemd systems it does, I guess. Not exactly sure what happens in the container. Anyway, cf-execd/cf-agent should see the cf-serverd process running even without a sleep.

Reply:
@vpodzime Okay, 🤞